### DISCUSSION TITLE
Crossplane behaviour upon deleting the whole control plane of Workload cluster.

### DISCUSSION THREAD

**Discussion by Sir-gits-a-lot** [crossplane/crossplane#4646] [Q&A]
If I launch a Workload cluster using crossplane,(say EKs) later go to the management console and delete the whole cluster itself including the  control plane itself. 
Crossplane is actually able to bring back the cluster but I wanna know how because if control plane of a cluster is deleted, we loose the kube-api server itself. 

Also, what happens to Workload running in that cluster. Is it backed up somewhere outside of the Workload cluster. If not then when crossplane reconciled the cluster will it also have the pods that were deployed by the user in that cluster?

Does anybody have clear idea on this ?

**Comment by bobh66** [comment]
Since Crossplane is creating the Workload cluster from scratch, including the control plane/API server (the EKS Cluster resource, for example) then re-creating that resource when it is manually deleted is the same as creating it in the first place.  The API server of the target cluster is just another resource that Crossplane "knows" how to create.

As for the applications/pods/etc running in that Workload cluster, they are not backed up anywhere unless you have done that as part of your Composite.   Anything that you deploy into the cluster from Crossplane (provider-helm Releases, provider-kubernetes Objects, etc) will be recreated (or at least will attempt to recreate) in the new cluster when it is available.

The exact behavior will be highly dependent on the specific workload that you have deployed into the cluster.

**Reply by dmlayton** [reply]
Why is it specific to the workload? I might have thought it was specific to a provider. Thank you for the insights.

**Comment by dmlayton** [comment]
@Sir-gits-a-lot what did you do in the end? I'm in the same situation. Clearly, the controlplane cluster might go down. Furthermore, I'd rather just tear down and spin up the controlplane cluster than to try and repair it if I get into some kind of dependency hell or undesirable state. I mean it seems like any time I want to delete something I have an issue with finalizers. Also an upgrade could easily go bad. 

It can't be that now all my manged resources are orphaned for the end of time. It seems like if I have a map from the controlpane identifiers (what us that Kind and Name) to the providers identifiers for every resource catalogued somewhere outside the controlplane, I should be able to generically un-orphan all the resources. IS this not generic and easy with Crossplane? IF not, how do people actually use it in the real world?

**Comment by dmlayton** [comment]
I'll update there as well. It seems like the options are:
update the desired state wherever it is mastered with the external-label that the managed resource gets after it is provisioned and make sure when it gets applied after recovery
backup the entire etcd as described here, https://aws.amazon.com/blogs/opensource/disaster-recovery-when-using-crossplane-for-infrastructure-provisioning-on-aws/ with something  like Velero
Use Upbound which has this as a feature <- I suspect it uses option 2

The only issue with option 1 is that if you are using claims or compositions to express your desired state then you don't have a declaration at the resource level and the external-label values are, of course, determined at a resource level by the external provider -- e.g. AWS.
